cfpb/design-system-react
# [Dropdown] Updated stylings, states and stories

Closes #173 

## Changes

- [New] Uses the standard CFPB accessibility styling (dotted outline when focused) in the Options 
- [New] Utilizes the proper  CFPB-style 'down' [&#9660;](http://www.fileformat.info/info/unicode/char/25bc/index.htm) icon in the Dropdown
- [New] `error` prop that causes a red border around the Dropdown
- [New] Dropdown Storybook story ('With Error')
- [Update] Utilize the current library's Checkbox instead of a generic checkbox
- [Update] Removed internal useState usage in **_both_** Checkbox and Dropdown. State must come externally in a `controlled` manner.
- [Update] Update the storybook stories of Checkbox and Dropdown to use external state.
- [Update] Update TypeScript settings
- [Update] Update the Multi-Select variation
- [Update] Incorporate the Menu Options styling
- [Update] Incorporate the styling from Figma

## How to test this PR

1. `yarn vitest`
2. See that everything passes
3. `yarn storybook`
4. Test the Checkbox and Dropdown stories
5. Inspect the new styling and compare with the figma and CFPB design

## Screenshots

### Default
<img width="1119" alt="Screenshot 2023-09-19 at 4 01 28 PM" src="https://github.com/cfpb/design-system-react/assets/13324863/eb1572a1-28af-4e42-8bfa-4abbfeaac331">

### Error
<img width="1154" alt="Screenshot 2023-09-19 at 4 01 09 PM" src="https://github.com/cfpb/design-system-react/assets/13324863/f6d47906-6814-44af-8b1a-95306ec60c62">

### Multiselect with Pills and Checkboxes
<img width="1106" alt="Screenshot 2023-09-19 at 4 02 01 PM" src="https://github.com/cfpb/design-system-react/assets/13324863/d400d2c9-486e-4475-81f7-d7386294ee0a">

### Multiselect -- Menu Options (Dotted Outline While Focused)
<img width="1200" alt="Screenshot 2023-09-25 at 9 53 40 AM" src="https://github.com/cfpb/design-system-react/assets/13324863/aa6bb7f2-fa88-422a-b198-f11260b0a07a">
